### PR TITLE
Find record using pointer from error

### PR DESCRIPTION
**Pull request by gniezen**
When the platform returns an error, there is an index pointer to the specific record in the block sent during the API call. This PR finds that offending record using the pointer.

### CODE REVIEW

File: lib/core/api.js [hunk 1]
```diff
@@ -392,7 +392,16 @@ function addDataToDataset(data, datasetId, blockIndex, uploadType, callback) {
       if(status.meta) {
         err.requestTrace = status.meta.trace.request;
       }
-      api.log(JSON.stringify(status));
+      api.log(JSON.stringify(status, null, '\t'));
+      if(status.errors && status.errors.length > 0) {
+        for (var i in status.errors) {
+          var hpattern = /\/(\w+)\//;
```
Comment (darinkrauss) [inline lib/core/api.js]: Should this be `\d` not `\w`?

File: lib/core/api.js [hunk 2]
```diff
@@ -392,7 +392,16 @@ function addDataToDataset(data, datasetId, blockIndex, uploadType, callback) {
...
+          var toMatch = hpattern.exec(status.errors[i].source.pointer);
+          if (toMatch[1]) {
+            api.log('Offending record for error',i,':', JSON.stringify(data[parseInt(toMatch[1])], null, '\t'));
```
Comment (darinkrauss) [inline lib/core/api.js]: Space between parameters?

File: lib/core/api.js (hunk 1 above)
Comment (gniezen) [inline lib/core/api.js]: Yep, that will be better.

**Comment by gniezen** [comment]
As this change is not user-facing, I'm going to merge this without a version bump and it will form part of the next release (which will most likely be the next iteration of the Medtronic bugfix).

**Pull request merged**